csrc/predicate_compute.h

```diff
@@ -31,6 +31,19 @@ class PredicateCompute {
   static Val* getElectSyncPredicate(
       kir::Predicate* pred,
       const std::vector<ForLoop*>& loops);
+
+  //! Used in two cases:
```

Comment:
I feel like these two types of predicates deserve for their own `PredicateType`.  That seems to make the logic a little more explicit and clear.

Reply:
decomposed to two types ` OneDimTmaLoadExpectArrive ` and ` OneDimTmaWaitParity`